compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/BIRPackageSymbolEnter.java

```diff
@@ -1744,6 +1749,17 @@ public BType readType(int cpI) throws IOException {
             return null;
         }
 
+        private void readFieldAnnotations(BTypeSymbol typeSymbol) throws IOException {
+            String fieldName = getStringCPEntryValue(inputStream);
+            var fieldFlags = inputStream.readLong();
+            byte[] docBytes = readDocBytes(inputStream);
+            BType fieldType = readTypeFromCp();
+            BVarSymbol varSymbol = new BVarSymbol(fieldFlags, Names.fromString(fieldName), typeSymbol.pkgID, fieldType,
+                                                  typeSymbol, symTable.builtinPos, COMPILED_SOURCE);
+            defineMarkDownDocAttachment(varSymbol, docBytes);
+            typeSymbol.annotations = varSymbol;
```

Comment:
What is this? We need to individually set the annotations to each field. Are you setting the entire map to the type symbol? That is wrong.

Reply:
Check how function parameters are handled for example. The map is only for runtime.

Reply:
@ushirask added the member-level annotation logic already - https://github.com/ballerina-platform/ballerina-lang/blob/74e19c8c6997735d5009f87964e73eed490a558a/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/BIRPackageSymbolEnter.java#L1607.

Reply:
Updated.